src/components/shared-panel/shared-panel.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import {
+  EmailShareButton,
+  FacebookMessengerShareButton,
+  FacebookShareButton,
+  TwitterShareButton,
+  WhatsappShareButton,
+} from 'react-share';
+import styled from 'styled-components';
+import { up } from '../breakpoint/breakpoint';
+import { IconEmail } from '../icons/social/email';
+import { IconFacebook } from '../icons/social/facebook';
+import { IconMessenger } from '../icons/social/messenger';
+import { IconTwitter } from '../icons/social/twitter';
+import { IconWhatsapp } from '../icons/social/whatsapp';
+
+export interface AuthorCredentialsProps {
+  facebookId: string;
+}
+
+const SharedPanelContainer = styled.div`
+  color: #668cff;
+  margin-top: 26px;
+  font-weight: bold;
+  font-size: 16px;
+`;
+
+const SharedPanelText = styled.p`
+  font-weight: bold;
+  font-size: 16px;
+`;
+
+const SocialLinks = styled.ul`
+  all: unset;
+  order: 2;
+
+  ${up('md')} {
+    order: 1;
+    margin-bottom: 20px;
+  }
+`;
+
+const SocialLinkItem = styled.li`
+  display: inline-block;
+  margin-right: 16px;
+  cursor: pointer;
+  &:last-of-type {
+    margin-right: 0;
+  }
+`;
+
+export const SharedPanel: React.FC<AuthorCredentialsProps> = ({
+  facebookId,
+}) => {
+  const shareUrl = typeof window !== `undefined` ? window.location.href : '';
```

Comment:
instead of `typeof window !== `undefined`` you can also use `process.browser ? ...`. it's a bit more readable

Reply:
thanks for the idea... will take a look

UPD:
sorry, I have objection:

`process.browser` seems to be  defined by webpack and I am not sure we can rely on it. 

not sure if I understand clearly, how it save the day:
> It also exposes a "browser" member (i.e. process.browser) which is true in this implementation but undefined in node. This can be used in isomorphic code that adjusts it's behavior depending on which environment it's running in.

and `typeof window !== undefined` is not too ugly to me